### PR TITLE
8299340: CreateProcessW lpCommandLine must be mutable

### PR DESCRIPTION
**Pull request by RogerRiggs**
Launching of processes on Windows using `ProcessCreateW` with a Unicode character set requires the buffer to be writable. An access violation might occur if `ProcessCreateW` writes to the command line string. The current implementation fetches the command line string using JNI GetStringChars returning a buffer that should not be modified. The code is unchanged since 2015.  There have not been any reported faults in that time.

This change copies the command line to a separately allocation mutable buffer to satisfy the Windows requirement.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299340](https://bugs.openjdk.org/browse/JDK-8299340): CreateProcessW lpCommandLine must be mutable

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13894/head:pull/13894` \
`$ git checkout pull/13894`

Update a local copy of the PR: \
`$ git checkout pull/13894` \
`$ git pull https://git.openjdk.org/jdk.git pull/13894/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13894`

View PR using the GUI difftool: \
`$ git pr show -t 13894`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13894.diff">https://git.openjdk.org/jdk/pull/13894.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13894#issuecomment-1546073817)

### CODE REVIEW

File: src/java.base/windows/native/libjava/ProcessImpl_md.c [hunk 1]
```diff
@@ -379,13 +380,21 @@ Java_java_lang_ProcessImpl_create(JNIEnv *env, jclass ignored,
...
-                        ret = processCreate(
-                            env,
-                            pcmd,
-                            penvBlock,
-                            pdir,
-                            handles,
-                            redirectErrorStream);
+                        // Copy command line to mutable char buffer; CreateProcessW may modify it
+                        jsize cmdLen = (*env)->GetStringLength(env, cmd);
+                        WCHAR *pcmdCopy = (WCHAR*)malloc(cmdLen * sizeof(WCHAR));
```
Comment (naotoj) [inline src/java.base/windows/native/libjava/ProcessImpl_md.c:385]: Should this include null terminator, as it is interpreted as `LPWSTR` which is null-terminated?

Comment (RogerRiggs) [reply src/java.base/windows/native/libjava/ProcessImpl_md.c:385]: It seems like it should now.  Java strings are not null terminated and any null termination would have accidental.
But it has worked for a long time (2013) without explicit Unicode null termination.

**Review by naotoj — APPROVED** [review]
LGTM

**Pull request closed without merging**